appsec-kit-starter/src/test/java/com/vaadin/appsec/service/AppSecDevToolsPluginTest.java

```diff
@@ -0,0 +1,156 @@
+        appSecService.when(AppSecService::getInstance)
+                .thenReturn(appSecServiceInstance);
+    }
+
+    @AfterEach
+    public void cleanup() {
+        appSecService.close();
+    }
+
+    @Test
+    public void handleConnect_sendsCommands() {
+        when(appSecServiceInstance.getNewVulnerabilities())
+                .thenReturn(Collections.emptyList());
+        when(appSecServiceInstance.addScanEventListener(any()))
+                .thenReturn(() -> {
+                });
+
+        appSecDevToolsPlugin.handleConnect(devToolsInterface);
+
+        verify(devToolsInterface, times(2)).send(anyString(), any());
+    }
+
+    @Test
+    public void handleConnect_addsScanEventListener() {
+        when(appSecServiceInstance.getNewVulnerabilities())
+                .thenReturn(Collections.emptyList());
+        when(appSecServiceInstance.addScanEventListener(any()))
+                .thenReturn(() -> {
+                });
+
+        appSecDevToolsPlugin.handleConnect(devToolsInterface);
+
+        verify(appSecServiceInstance, times(1)).addScanEventListener(any());
+        assertEquals(1, appSecDevToolsPlugin.scanEventRegistrations.size());
+    }
+
+    @Test
+    public void handleConnect_doesntAddScanEventListener_ifAlreadyAdded() {
+        when(appSecServiceInstance.getNewVulnerabilities())
+                .thenReturn(Collections.emptyList());
+        appSecDevToolsPlugin.scanEventRegistrations.put(devToolsInterface,
+                () -> {
+                });
+
+        appSecDevToolsPlugin.handleConnect(devToolsInterface);
+
+        verify(appSecServiceInstance, never()).addScanEventListener(any());
+        assertEquals(1, appSecDevToolsPlugin.scanEventRegistrations.size());
+    }
+
+    @Test
+    public void scanEventListener_sendsScanResult() {
+        var appSecScanEvent = mock(AppSecScanEvent.class);
+        when(appSecScanEvent.getNewVulnerabilities())
+                .thenReturn(Collections.emptyList());
+        when(appSecServiceInstance.getNewVulnerabilities())
+                .thenReturn(Collections.emptyList());
+
+        appSecDevToolsPlugin.handleConnect(devToolsInterface);
+
+        verify(appSecServiceInstance)
+                .addScanEventListener(appSecScanEventListenerCaptor.capture());
+        var appSecScanEventListener = appSecScanEventListenerCaptor.getValue();
+        appSecScanEventListener.scanCompleted(appSecScanEvent);
+
+        verify(devToolsInterface, times(3)).send(anyString(), any());
+    }
+
+    @Test
+    public void handleMessage_returnsTrue() {
+        var result = appSecDevToolsPlugin.handleMessage("command",
+                Json.createObject(), devToolsInterface);
+        assertTrue(result);
+    }
+
+    @Test
+    public void handleDisconnect_removesScanEventListener() {
+        var registration = mock(Registration.class);
+        appSecDevToolsPlugin.scanEventRegistrations.put(devToolsInterface,
+                registration);
```

Comment:
If we returned a mocked `Registration` from `AppSecService` (as suggested in a previous comment) we can then verify here that it is removed (no need to access `scanEventRegistrations`).

Reply:
Removed this.